Translate Fragments (#13)
reactjs/pt-BR.reactjs.org: content/docs/fragments.md

---
id: fragments
title: Fragmentos
permalink: docs/fragments.html
---

Um padrão comum no React é que um componente pode retornar múltiplos elementos. Os Fragmentos permitem agrupar uma lista de filhos sem adicionar nós extras ao DOM.

```js
render() {
  return (

>     <React.Fragment>

Reviewer: @gustavoaz7 blocos de código... não precisam ser traduzidos... somente comentários ;)

      <FilhoA />
      <FilhoB />
      <FilhoC />
    </React.Fragment>
  );
}
```

> Há também uma nova [sintaxe curta](#short-syntax) para declará-las, mas ainda não é suportada por todas as ferramentas populares.

Reviewer: ```suggestion
Há também uma nova [sintaxe curta](#short-syntax) para declará-las. Porém, ainda não é suportada por todas as ferramentas populares.
```

## Motivação {#motivation}

Um padrão comum é para um componente retornar uma lista de filhos. Considerando o código React a seguir:

```jsx
class Table extends React.Component {
  render() {
    return (
      <table>
        <tr>

>           <Colunas />

Reviewer: ```suggestion
<Columns />
```

procura manter os blocos de código em inglês ;)

        </tr>
      </table>
    );
  }
}
```

> `<Colunas />` precisaria retornar múltiplos elementos `<td>` para que o HTML renderizado fosse válido. Se um div pai for usado dentro do `render()` de `<Colunas />`, então o HTML resultante será inválido.

Reviewer: @gustavoaz7 mesma coisa aqui. Seguindo o procedimento da issue #4

```jsx

> class Colunas extends React.Component {

Reviewer: @gustavoaz7 mesmo item - não precisa ser traduzido.

</table>
```

Os Fragmentos resolvem este problema.

## Uso {#usage}

```jsx{4,7}

> class Colunas extends React.Component {

Reviewer: @gustavoaz7 mesma coisa...

</table>
```

### Sintaxe curta {#short-syntax}

Existe uma sintaxe nova e mais curta que você pode usar para declarar fragmentos. Parecem tags vazias:

```jsx{4,7}

> class Colunas extends React.Component {

Reviewer: @gustavoaz7 aqui tbm

}
```

Você pode usar `<></>` da mesma forma que você usaria qualquer outro elemento, exceto que ele não suporta chaves ou atributos.

> Observe que **[muitas ferramentas ainda não o suportam](/blog/2017/11/28/react-v16.2.0-fragment-support.html#support-for-fragment-syntax)** então é melhor escrever `<React.Fragment>` até que as ferramentas passem a suportá-lo.

Reviewer: ```suggestion
Observe que **[muitas ferramentas ainda não possuem suporte](/blog/2017/11/28/react-v16.2.0-fragment-support.html#support-for-fragment-syntax)** então é melhor escrever `<React.Fragment>` até que as ferramentas passem a ter suporte.
```
